presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -36,14 +39,34 @@ public ThriftMetastoreModule(String connectorId)
     }
 
     @Override
-    public void configure(Binder binder)
+    protected void setup(Binder binder)
     {
         binder.bind(HiveMetastoreClientFactory.class).in(Scopes.SINGLETON);
         binder.bind(HiveCluster.class).to(StaticHiveCluster.class).in(Scopes.SINGLETON);
         configBinder(binder).bindConfig(StaticMetastoreConfig.class);
 
         binder.bind(HiveMetastore.class).to(ThriftHiveMetastore.class).in(Scopes.SINGLETON);
-        binder.bind(ExtendedHiveMetastore.class).annotatedWith(ForCachingHiveMetastore.class).to(BridgingHiveMetastore.class).in(Scopes.SINGLETON);
+
+        if (buildConfigObject(HiveClientConfig.class).getRecordingPath() != null) {
+            binder.bind(ExtendedHiveMetastore.class)
+                    .annotatedWith(ForRecordingHiveMetastore.class)
+                    .to(BridgingHiveMetastore.class)
+                    .in(Scopes.SINGLETON);
+            binder.bind(ExtendedHiveMetastore.class)
+                    .annotatedWith(ForCachingHiveMetastore.class)
+                    .to(RecordingHiveMetastore.class)
+                    .in(Scopes.SINGLETON);
+            newExporter(binder).export(ExtendedHiveMetastore.class)
+                    .annotatedWith(ForCachingHiveMetastore.class)
```

Comment:
- change `.export(..)`
- drop `. annotatedWith`
i.e.
```
newExporter(binder).export(RecordingHiveMetastore.class)
    .as(...
```

Reply:
One way (that seems a bit cleaner) is to add a binding for `RecordingHiveMetastore`:
```java
binder.bind(RecordingHiveMetastore.class).in(Scopes.SINGLETON);
newExporter(binder).export(RecordingHiveMetastore.class)
        .as(generatedNameOf(RecordingHiveMetastore.class, connectorId));

Reply:
Indeed it worked. Thanks!